tests/integration/run_modes/test_skip.py

```diff
@@ -0,0 +1,250 @@
+                'skip': {
+                    'outputs': 'failed'
+                }
+            }
+        }
+    })
+    schd = scheduler(wid, run_mode='live', paused_start=False)
+    assert await reftest(schd) == {
+        ('1/did_fail', ('1/should_fail',),),
+        ('1/foo', None,),
+        ('1/opt', None,),
+        ('1/optional_outs_produced', ('1/opt',),),
+        ('1/required_outs', ('1/foo', '1/foo',),),
+        ('1/should_fail', None,),
+        ('1/started_always', ('1/foo',),),
+        ('1/submitted_always', ('1/foo',),),
+        ('1/success_if_optional', ('1/foo', '1/foo',),),
+    }
+
+
+async def test_doesnt_release_held_tasks(
+    one_conf, flow, scheduler, start, log_filter
+):
+    """Point 5 of the proposal
+    https://github.com/cylc/cylc-admin/blob/master/docs/proposal-skip-mode.md
+
+    | Tasks with run mode = skip will continue to abide by the is_held
+    | flag as normal.
+
+    """
+    one_conf['runtime'] = {'one': {'run mode': 'skip'}}
+    schd = scheduler(flow(one_conf), run_mode='live')
+    async with start(schd) as log:
+        itask = schd.pool.get_tasks()[0]
+        msg = 'held tasks shoudn\'t {}'
+
+        # Set task to held and check submission in skip mode doesn't happen:
+        itask.state.is_held = True
+        schd.task_job_mgr.submit_task_jobs(
+            schd.workflow,
+            [itask],
+            schd.server.curve_auth,
+            schd.server.client_pub_key_dir,
+            run_mode=schd.get_run_mode()
+        )
+        assert not log_filter(log, contains='=> running'), msg.format('run')
+        assert not log_filter(log, contains='=> succeeded'), msg.format(
+            'succeed')
+        # Release held task and assert that it now skips successfully:
+        schd.pool.release_held_tasks(['1/one'])
+        schd.task_job_mgr.submit_task_jobs(
+            schd.workflow,
+            [itask],
+            schd.server.curve_auth,
+            schd.server.client_pub_key_dir,
+            run_mode=schd.get_run_mode()
+        )
+        assert log_filter(log, contains='=> running'), msg.format('run')
+        assert log_filter(log, contains='=> succeeded'), msg.format('succeed')
+
+
+async def test_force_trigger_doesnt_change_mode(
+    flow, scheduler, run, complete
+):
+    """Point 6 from the skip mode proposal
+    https://github.com/cylc/cylc-admin/blob/master/docs/proposal-skip-mode.md
+
+    | Force-triggering a task will not override the run mode.
+    """
+    wid = flow({
+        'scheduling': {'graph': {'R1': 'slow => skip'}},
+        'runtime': {
+            'slow': {'script': 'sleep 6'},
+            'skip': {'script': 'exit 1', 'run mode': 'skip'}
+        }
+    })
+    schd = scheduler(wid, run_mode='live', paused_start=False)
+    async with run(schd):
+        schd.pool.force_trigger_tasks(['1/skip'], [1])
+        # This will timeout if the skip task has become live on triggering:
+        await complete(schd, '1/skip', timeout=6)
```

Comment:
The workflow times out after 6 seconds, but the job takes 6 seconds. This will cause the test to become flaky (because there are 0 seconds of margin).